simple/reprovide.go

```diff
@@ -119,44 +142,27 @@ func (rp *Reprovider) Reprovide() error {
 	return nil
 }
 
-// Trigger starts reprovision process in rp.Run and waits for it
+// Trigger starts reprovision process in rp.Run and waits for it to finish.
+//
+// Returns an error if a reprovide is already in progress.
 func (rp *Reprovider) Trigger(ctx context.Context) error {
-	progressCtx, done := context.WithCancel(ctx)
-
-	var err error
-	df := func(e error) {
-		err = e
-		done()
+	doneCh := make(chan error, 1)
```

Comment:
the use of `doneCh` here as the channel to be given an error/closed indicating completion of a single reprovide trigger, as opposed to the `rp.doneCh` which is used to block the `Close` method on shutdown is an unfortunate overload of naming. Consider renaming one to indicate they aren't related.